### PR TITLE
net.vdrift.VDrift

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/VDrift/vdrift/pull/197
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
  - already upstreamed:
    - https://github.com/VDrift/vdrift/commit/35b047c8a2feffda7dfa7cc6bb2c32084683afd2
    - https://github.com/VDrift/vdrift/commit/2f1f72f2a7ce992b0aad30dc55509b966d1bb63d
    - https://github.com/VDrift/vdrift/commit/7d04c723a165109e015204642f4984a1a4452ccb
    - https://github.com/VDrift/vdrift/commit/21ee61aa5d78acb15f06d2934d80305a39bf6d54
    - https://github.com/VDrift/vdrift/commit/5e6758a21883ae3d13276afa2de314a1486c2049
  - submitted:
    - https://github.com/VDrift/vdrift/pull/197

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build net.vdrift.VDrift

File: net.vdrift.VDrift.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      - "*"
+    sources:
+      - type: archive
+        url: http://downloads.sourceforge.net/project/scons/scons/3.0.4/scons-3.0.4.tar.gz
+        sha256: e2b8b36e25492720a05c0f0a92a219b42d11ce0c51e3397a1e8296dfea1d9b1a
+    build-commands:
+      - python setup.py install --prefix=${FLATPAK_DEST} --optimize=1
+
+  - name: bullet
+    buildsystem: cmake
```
Comment (bbhtt) [inline net.vdrift.VDrift.yaml:29]:
```suggestion
    buildsystem: cmake-ninja
```

**Comment by hfiguiere** [comment]
```
Error: Error on line 19 char 8: Element “ul” was closed, but the currently open element is “description”
```

I guess you didn't build with the patches.

File: net.vdrift.VDrift.yaml (hunk 1 above)
Comment (Mailaender) [reply net.vdrift.VDrift.yaml:29]: Does not work.

**Comment by Mailaender** [comment]
The patches have line ending problems and `use-git: true` only seems to skip them.

File: net.vdrift.VDrift.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      - type: patch
+        path: appdata2.patch
+        use-git: true
+        options:
+          - --ignore-whitespace
+      - type: patch
+        path: appdata3.patch
+        use-git: true
+        options:
+          - --ignore-whitespace
```
Comment (hfiguiere) [inline net.vdrift.VDrift.yaml:87]:
```suggestion
      - type: patch
        paths:
          - appdata1.patch
          - appdata2.patch
          - appdata3.patch
        use-git: true
        options:
          - --ignore-whitespace
```

File: net.vdrift.VDrift.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      - type: file
+        path: vdrift-128x128.png
+      - type: patch
+        path: scons3.patch
+      - type: patch
+        path: scons3in.patch
+      - type: patch
+        path: bullet-double.patch
+      - type: patch
+        path: gcc6.patch
```
Comment (hfiguiere) [inline net.vdrift.VDrift.yaml:72]:
```suggestion
      - type: patch
        paths: 
          - scons3.patch
          - scons3in.patch
          - bullet-double.patch
          - gcc6.patch
```

File: net.vdrift.VDrift.yaml (hunk 2 above)
Comment (Mailaender) [reply net.vdrift.VDrift.yaml:87]: I tried that. Does not work.

**Comment by Mailaender** [comment]
bot, build net.vdrift.VDrift

File: net.vdrift.VDrift.yaml (hunk 2 above)
Comment (hfiguiere) [reply net.vdrift.VDrift.yaml]: define "does not work". I use that form quite often.

**Comment by Mailaender** [comment]
It crashes for me.

```
ERROR: Failed to load /home/matthias/.vdrift/VDrift.config
[...]
/usr/include/c++/13.2.0/bits/stl_vector.h:1125: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = float; _Alloc = std::allocator<float>; reference = float&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.
```

Sorry I was so busy with the broken patches I didn't notice that this can't be sandboxed.

**Comment by hfiguiere** [comment]
this crash is a bug in the program. It's crashing because we build with assert on while most build without to avoid crashes like that. Just to repeat it's a bug in the code.

I don't believe it has anything to do with sandboxing.

**Pull request closed without merging**